File: Math/0788.Rotated Digits/Solution.java
class Solution {
    public int rotatedDigits(int n) {
        Set<Integer> nums = new HashSet<>(), valid = new HashSet<>();
        for (int v : new int[]{0, 1, 2, 5, 6, 8, 9}) {
            nums.add(v);
        }
        for (int v : new int[]{2, 5, 6, 9}) {
            valid.add(v);
        }
        int ans = 0;
        for (int i = 0; i < n + 1; i++) {
            int cur = i;
            boolean flag = false;
            while (cur > 0) {
                if (!nums.contains(cur % 10)) {
                    break;
                }
                if (valid.contains(cur % 10)) {
                    flag = true;
                }
                cur /= 10;
            }
            if (cur <= 0 && flag) {
                ans++;
            }
        }
        return ans;
    }
}

File: Math/0788.Rotated Digits/Solution.py
class Solution:
    def rotatedDigits(self, n: int) -> int:
        nums = [0, 1, 2, 5, 6, 8, 9]
        valid = [2, 5, 6, 9]
        ans = 0
        for i in range(n + 1):
            cur = i
            flag = False
            while cur > 0:
                if cur % 10 not in nums:
                    break
                if cur % 10 in valid:
                    flag = True
                cur //= 10
            if cur <= 0 and flag:
                ans += 1
        return ans
